src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -2671,9 +2665,11 @@ theorem getMsbD_rotateLeft_of_lt {n w : Nat} {x : BitVec w} (hi : r < w):
       by_cases h₁ : n < w + 1
       · simp only [h₁, decide_true, Bool.true_and]
         have h₂ : (r + n) < 2 * (w + 1) := by omega
-        rw [mod_eq_sub_of_le_of_lt (by omega) (by omega)]
+        rw [Nat.mod_eq_sub_of_le_of_lt (n := 1)]
         congr 1
-        omega
+        · omega
+        · omega
+        · omega
```

Comment:
I think 
```
        rw [Nat.mod_eq_sub_of_le_of_lt (n := 1) (by omega) (by omega)]
        congr 1
        omega
```
is the best one can do here, or otherwise
```
        rw [Nat.mod_eq_sub_of_le_of_lt (n := 1)]
        congr 1
        all_goals omega
```

Reply:
Actually one could also use auto-params here:
```
theorem mod_eq_sub_of_le_of_lt {x w n : Nat} (x_ge : w * n ≤ x := by omega) (x_lt : x < w * (n + 1) := by omega) :
    x % w = x - w * n := by
  simp only [ge_iff_le, gt_iff_lt] at x_ge x_lt
```
and then just   
```
        rw [Nat.mod_eq_sub_of_le_of_lt (n := 1)]
        congr 1
        omega
```
works. But let's not do that as a one-off. (It may be interesting to think about using omega as a pervasive auto-param for side condition inequalities, however!)